src/main/java/duke/ChatSession.java

```diff
@@ -0,0 +1,102 @@
+package duke;
+
+import duke.task.Task;
+import java.util.ArrayList;
+import java.io.FileWriter;
+import java.io.IOException;
+
+public class ChatSession {
+    // Store subclasses of tasks
+    ArrayList<Task> taskList;
+
+    public ChatSession() {
+        this.taskList = new ArrayList<Task>();
+    }
+
+    public void startSession() {
+        System.out.println("____________________________________________________________");
+        System.out.println("Hello! I'm Duke");
+        System.out.println("What can I do for you?");
+        System.out.println("____________________________________________________________");
+    }
+
+    public void endSession() {
+        System.out.println("____________________________________________________________");
+        System.out.println("Bye. Hope to see you again soon!");
+        System.out.println("____________________________________________________________");
+    }
+
+    public void addInitialTask(Task task) {
+        taskList.add(task);
+    }
+
+    public void addTask(Task task) {
+        // Create a new Task, append to taskList
+        taskList.add(task);
+
+        // Add and save to offline file
+        try {
+            this.addToFile(task);
+        } catch (IOException e) {
+            System.out.println("Something went wrong: " + e.getMessage());
+        }
+
+        // Return the added task
+        System.out.println("____________________________________________________________");
```

Comment:
Consider extracting repeated strings like this into constants for better reusability.